### DISCUSSION TITLE
lifeline reports

### DISCUSSION THREAD

**Discussion by juntiedt2** [zwave-js/node-red-contrib-zwave-js#151] [Q&A]
My z-wave thermostat sends 8 different reports to lifeline. One of it is Battery report. I can configure the report to be send once a day. If I do this through the UI it immedeately sends a "get value response". If I do battery report config through the command factory I don't get a "get value response". Why?

And I don't  get a report once a day as well..

If I want to get the lifeline reports do I have to associate the thermostat with Group 1? Could this be my prblem?

Z-Wave Config DB:
[https://devices.zwave-js.io/?jumpTo=0x0148:0x0003:0x0001:0.0](https://devices.zwave-js.io/?jumpTo=0x0148:0x0003:0x0001:0.0)

**Comment by marcus-j-davies** [comment]
Hey! @juntiedt2.

The UI does calls to `getValue`
And the calling entity (in this case the UI) will get the result (via a `GET_VALUE_RESPONSE`),
but at the same time, this event is passed to any listening device node.

if you open **UI Monitor** - you can see all the getValue(s) that are being called, it does this to fill the values, so you can see them in the UI.

The response being passed back to the UI.
https://github.com/zwave-js/node-red-contrib-zwave-js/blob/b14fef607fe2cd9015a146d66e33b48d7cead2ec/zwave-js/zwave-js.js#L1347

The response also being distributed to any potential device node(s) 
https://github.com/zwave-js/node-red-contrib-zwave-js/blob/b14fef607fe2cd9015a146d66e33b48d7cead2ec/zwave-js/zwave-js.js#L1385

It's probably something I can improve at some point, but currently low priority.

--------------------------------------

The life line group in zwave spec is designed to receive everything (and its should only be the controller that is associated to it)

devices will have at least 1 group that is designated `Lifeline` - i.e the destination controller for all reports.
a device can have many many groups - all having some different use.

Associations/Groups are different on every device.
ideally, the the Lifeline should receive them all, but again, it depends on what the user manual states.

if a certain report, is not delivered via its lifeline (but does have a group for it)
you may be able to also associate the controller to it.


Associations / Endpoints are always a mess in zwave - Manufacturer A does one thing,  Manufacturer B does something completely opposite.


If you fancy some reading
https://zwave-js.github.io/node-zwave-js/#/getting-started/migrating-to-v8?id=reworked-handling-of-endpoints-and-lifeline-associations

**Reply by marcus-j-davies** [reply]
> And I don't get a report once a day as well..

If you are not getting reports, then they are not being transmitted over the lifeline association.
or the device is using a different endpoint for them.

Our configs (per device) can work around these oddities, so if you are not getting any reports, we will need to see an interview log of that device, so the config file for it, can be adjusted.

Edit:
There might actually be a fix for this already.
Let me know what is happening, and I'll know more.

**Reply by juntiedt2** [reply]
Ok, part one is solved.
Reports are still open. 
How can I get the interview log. Do you mean UI Monitor?
next question: are the devices automaticaly associated with the controller (lifeline)?
what is for example the command class for Battery report - 64?

**Reply by marcus-j-davies** [reply]
Hi @juntiedt2,

> How can I get the interview log

Turn on logging in the controller config (don't forget to deploy after) , and then hit 'Interview Node', for the selected node.

> are the devices automatically associated with the controller (lifeline)

Yes - this is trivial, devices advertise what is the life line group, and the controller will be assigned to it.
All reports are sent to the life line group (well the device should), you can see all associations in the Manage Associations UI for the selected node - Group 1 should be allocated to at least the controller (1)

> what is for example the command class for Battery report - 64?

No, 64 (0x40) is the ID for Thermostat Mode CC
The battery command class ID is 128 (0x80).

Before anything however...

I am trying to find out why you don't get the timed battery report (or any other timed reports?)
can you be certain of that - how did you come to the conclusion, are you capturing everything coming in from that device, and storing it somewhere - and don't see anything over a 24 hour period?

If you really are not getting any reports, then:

1 - The lifeline is not correct on the device, and the device has advertised the wrong one. OR
2 - A recent update to the driver, has implemented fixes for this device.

It's worth noting, this device has known firmware bugs, that may extend to your issue.
see this: https://github.com/zwave-js/node-red-contrib-zwave-js/discussions/140

if you are not getting any timed reports, we first try an update of the driver.
if still nothing, we need a log of the interview - to see what the device is telling us about its endpoints.

**Comment by juntiedt2** [comment]
ok, I will log command class 128 for 25 hours and see what happens.
btw, I am logging my wake ups to my mobile phone via Telegram. Don' t need the computer to check :-)

**Reply by marcus-j-davies** [reply]
I would actually log these events (regardless of CC)

**VALUE_UPDATED**,  **NOTIFICATION**, **VALUE_NOTIFICATION**

therefore you have a 25 hour window of every thing possible.
Note: The database is updated when values are received.

**Reply by juntiedt2** [reply]
I have logged all messages for 2 days and no life line reports came up.
How can I check if the thermostat sends something to the controller?
Could it use a different Group?

**Reply by marcus-j-davies** [reply]
Hi @juntiedt2 

This device only has 1 group (lifeline) according to its manual. In the Associations UI, if its there, then everything lifeline based should be setup.

So, if above checks out:

**Fix1**: Update Driver (it may have a fix for this device)
 - Navigate to your **.node-red** directory (_usually_ **/home/pi/.node-red**)
 - do **npm install zwave-js@latest**
 - Restart Node-Red, and once all back up, re-interview this device, and check over a 24 hour window if it has gotten things working.
    - you can use the UI to do an interview under the node options
 
 
 If Fix 1 - did not help:
 
 **Fix 2**: This might be a device that needs some config changes.

Our configs have certain flags that can be applied - specifically for this reason, but we only know about them, when issues occur.

Here is what to do.
 - Turn on Logging (double click controller Node), and turn on **File** Logging to Debug (deploy after)
 - Start a new interview on that device (so we can capture it in the log file) 
     - you can use the UI to do an interview under the node options
 - Wait for Interview to complete (should take a minute or 2)
 - Turn Logging off, and attach the log file here (drag) - no sensitive info is in there.
 
 With this log file, we can then adjust the config, so it works with this device

Basic Lifeline reports _should_ be trivial - but some devices need a little more focus put on them

**Reply by juntiedt2** [reply]
Hi Marcus,
Fix 1 did not work.
Please find attached requested log file.

[zwave-js.log](https://github.com/zwave-js/node-red-contrib-zwave-js/files/7580694/zwave-js.log)

**Reply by juntiedt2** [reply]
Docu of the thermostat
[Spirit_Z-Wave_Plus_Installation-and-Operation-Guide_web (1).pdf](https://github.com/zwave-js/node-red-contrib-zwave-js/files/7580773/Spirit_Z-Wave_Plus_Installation-and-Operation-Guide_web.1.pdf)

**Reply by marcus-j-davies** [reply]
Thanks @juntiedt2,

Ok, so with the log, we will now see if this device needs some modifications to its config.

@AlCalzone,
This device has done  the rounds, its been mentioned here previously with another environment, and a (now closed) reported issue on the driver repo - although I don't think its related.

The driver is not receiving any lifeline reports, but their only appears to be 1 group (lifeline).
that association is setup as per interview.

1 example is battery reports, not being received (i think there are a few others)

DB Link
https://devices.zwave-js.io/?jumpTo=0x0148:0x0003:0x0001:0.0
PDF + Interview Log attached above.

I'm guessing some flag/change is needed for its config?

**Reply by juntiedt2** [reply]
yes, all reports are missing. You will find the list of reports under Association in the Docu.

**Reply by AlCalzone** [reply]
I do see one unsolicited report:
```
2021-11-22T11:39:17.791Z DRIVER « [Node 013] [REQ] [BridgeApplicationCommand]
                                  └─[MultilevelSensorCCReport]
                                      type:  Air temperature
                                      scale: Celsius
                                      value: 19.1
```

There were a couple of commands that could not be sent during the interview, probably interference issues. Maybe the device itself is experiencing the same problems when it tries to report.

**Reply by juntiedt2** [reply]
I have 3 thermostats. All 3 do not send unsolicited reports once every 24 hours. 
Air temp reports depending on temp differences of 0.1 degree are working ok. (VALUE_UPDATED)
As I do not receive the unslicited reports I send a get value to the Battery Stats every 12 hours and at start up. This works ok as well.

**Reply by marcus-j-davies** [reply]
Hi @juntiedt2 

> Air temp reports depending on temp differences of 0.1 degree are working ok. 

i.e an unsolicited report 😄 
I thought you were not getting any? 

The fact you get updates when you are not asking for them (aka unsolicited) suggests that life line is working on this device.

therefore, if its simply the battery reports you are net getting, can you confirm you have enabled them todo so?
if you have, maybe pull the battery out and re-insert them.

I have motion sensors, that don't report battery on their intervals after an interview - and I need to pull the battery and re-insert it before the device behaves correctly.

if this still does nothing, this one maybe for the manufacture to answer 🤷‍♂️

**Reply by juntiedt2** [reply]
mea culpa! mea maxima culpa!
I thought that all reports are unsolicited once a day. Did not read good enough :-(
The thermostas are working. So far so good.
But I was not able to get all listed life line reports yet.
I implemented a few logging routines and I will check the results over the next few days. 
Many thanks for your support.
I will keep you updated.

**Comment by marcus-j-davies** [comment]
No worries,

The _unsolicited_ reports (where we have not asked for the  info) I believe you will get are (at least according to the devices manual - they are not always accurate):

 - Battery Reports (If enabled)
  - Thermo Mode (if changed on the device)
  - Set point (if changed on device)
  - Multi Level Sensor (Air Temp Changes)
  - Multi Level Switch (manual Valve position changes)
 
 All these will be delivered to the lifeline on that device (i.e. the controller - its configured during interview).
 
 if you have enabled battery reports, pull the battery then re-insert it, some of my motions sensors, stop reporting after the lifeline has been configured, only working again when the battery is re-inserted (I think its just a characteristic of certain devices).
 
 also check some of this 
 they might not be enabled 👍 
 
 
![image](https://user-images.githubusercontent.com/55892693/143473435-82b49ead-501c-48ff-81ce-4f53afce1c60.png)

**Reply by juntiedt2** [reply]
hi Marcus,
after testing for a while I have the folowing status:
Measured temp report is working ok
Valve Opening report is working ok
Battery report is not working. I now do a "value get" once a day at 1:00 and I am getting data back. They did not change until now. Will see what happens after a month or longer..
I have send an email to EUROTRONIC TECHNOLOGY GMBH. No response.
in the internet I found some posts which had the same problems with this thermostat.
If you are interested I would send you the n-r flow.

**Comment by marcus-j-davies** [comment]
Hi @juntiedt2 

 - Measured temp report is working ok
 - Valve Opening report is working ok
 
These will be delivered via the same lifeline  as the battery reports (or at least the battery reports should be).

it's very possible the battery reports are broken on this device, and use an endpoint that is not advertised from the device during its interview, or the device simply is not sending battery reports what so ever - Have you done a battery pull and reinsert - I have devices that don't send battery reports unless I pull and re-insert after an interview.
 
Given your have found other reports of this nature (and we have had Firmware Bugs on this device with other users), I'm lead to believe, it is indeed some buggy firmware that it is running.

further - you can query the battery maually (or in your case at a schedule), and it returns with no problem - so Battery reports should be sent automatically without issue.

My stance on this is, the device is simply not sending scheduled battery reports by its self, have they fixed this with a FW update?, did they get the manual wrong?

🤷‍♂️ 

Given you can still schedule a report your self, I suppose is a silver lining - but I agree, in that a message to EUROTRONIC, may not hurt.

**Comment by marcus-j-davies** [comment]
> If you are interested I would send you the n-r flow.

Sure, if I see something odd - I'll let you know.

**Reply by juntiedt2** [reply]
my flow:
[eurotronic SPIRIT Z-WAVE PLUS.txt](https://github.com/zwave-js/node-red-contrib-zwave-js/files/7655590/eurotronic.SPIRIT.Z-WAVE.PLUS.txt)
w

**Reply by marcus-j-davies** [reply]
Nice,

Quick question.
how are you concluding you don't get battery?
if its becuase you don't see it changing on your dashboard - it may not have had a capacity drop? 

Are you capturing/storing everything from the  controller side?

The only storing I see in your flow is here out to a file.

![image](https://user-images.githubusercontent.com/55892693/144743239-52bb3605-56b2-4080-9b26-73e9814c48e1.png)

But your **all Notifications** filter - does not include **VALUE_UPDATED** which I think the battery will fall under.
its good you don't have any VIDs in there (hence it will allow it all through), but if this is used to check if battery reports are coming in, you may need **VALUE_UPDATED**.

just a thought, but if you have other means - then carry on 😄

**Reply by juntiedt2** [reply]
Here is my loggin
[z-wave logging.txt](https://github.com/zwave-js/node-red-contrib-zwave-js/files/7655723/z-wave.logging.txt)
g:
12 13 and 14 are my thermostats

**Reply by juntiedt2** [reply]
if it is a unsolicited report once a day a capacity drop should not be relevant for the report

**Reply by marcus-j-davies** [reply]
Yup,

it was more if you were monitoring the dashboard for changes, but since you log everything, it should show up.
still, if its not there - I guess still waiting to see what  EUROTRONIC says.

I also found forum discussions mentioning this device and its lack of battery reports.
